cmd/metal-api/internal/metal/provisioning.go

```diff
@@ -100,42 +134,40 @@ func (p provisioningEventSequence) containsEvent(event ProvisioningEventType) bo
 	return result
 }
 
-func (e *ProvisioningEvent) hasExpectedSuccessor(log *zap.SugaredLogger, actualSuccessor ProvisioningEventType) bool {
-	currentEvent := e.Event
-
-	expectedSuccessors, ok := expectedSuccessorEventMap[currentEvent]
-	if !ok {
-		log.Errorw("successor map does not contain an expected successor for event", "event", currentEvent)
-		return false
+func postEvent(currentFSM *fsm.FSM, event ProvisioningEventType) (*fsm.FSM, error) {
+	var nextFSM *fsm.FSM
+	err := currentFSM.Event(string(event))
+	if err != nil && err.Error() != "no transition" {
```

Comment:
Checking for specific errors can be done with `errors.Is` or `errors.As` (https://pkg.go.dev/errors#Is), string comparison is error-prone. The state machine library returns a typed `NoTransition` error.